vip-helpers/class-user-cleanup.php

```diff
@@ -0,0 +1,115 @@
+<?php
+
+namespace Automattic\VIP\Helpers;
+
+// If the constant is set, don't run the cleanup
+if ( defined( 'VIP_SKIP_USER_CLEANUP' ) && true === VIP_SKIP_USER_CLEANUP ) {
+	add_filter( 'vip_do_user_cleanup', '__return_false' );
+}
+
+class User_Cleanup {
+	public static function parse_emails_string( $emails_string ) {
+		$emails = [];
+
+		$emails_string = trim( $emails_string );
+
+		if ( false !== strpos( $emails_string, ',' ) ) {
+			$emails_raw = explode( ',', $emails_string );
+		} else {
+			$emails_raw = [ $emails_string ];
+		}
+
+		foreach ( $emails_raw as $email_raw ) {
+			$email_raw = trim( $email_raw );
+
+			$email_filtered = filter_var( $email_raw, FILTER_VALIDATE_EMAIL );
+
+			if ( empty( $email_filtered ) ) {
+				continue;
+			}
+
+			$emails[] = $email_filtered;
+		}
+
+		return $emails;
+	}
+
+	public static function split_email( $email ) {
+		list( $email_username, $email_hostname ) = explode( '@', $email );
+
+		// Strip off everything after the +, if it exists
+		list( $email_username, ) = explode( '+', $email_username );
+
+		return [ $email_username, $email_hostname ];
+	}
+
+	public static function fetch_user_ids_for_emails( $emails ) {
```

Comment:
Thoughts on having this work in a strict match and wildcard modes?

It's not inconceivable for someone to try to use it to remove a specific email only to find out that other accounts were affected too.

Reply:
Going to pass on this. Since this is meant for offboarding, I think it's better to do a more thorough search. If someone wants to remove a specific user, they should use `wp user delete`.